compiler/rustc_abi/src/layout.rs

```diff
@@ -54,6 +54,36 @@ pub enum LayoutCalculatorError<F> {
 
     /// A union had no fields.
     EmptyUnion,
+
+    /// The fields or variants have irreconcilable reprs
+    ReprConflict,
+}
+
+impl<F> LayoutCalculatorError<F> {
+    pub fn without_payload(&self) -> LayoutCalculatorError<()> {
+        match self {
+            LayoutCalculatorError::UnexpectedUnsized(_) => {
+                LayoutCalculatorError::UnexpectedUnsized(())
+            }
+            LayoutCalculatorError::SizeOverflow => LayoutCalculatorError::SizeOverflow,
+            LayoutCalculatorError::EmptyUnion => LayoutCalculatorError::EmptyUnion,
+            LayoutCalculatorError::ReprConflict => LayoutCalculatorError::ReprConflict,
+        }
+    }
+
+    /// Format an untranslated diagnostic for this type
+    ///
+    /// Intended for use by rust-analyzer, as neither it nor `rustc_abi` depend on fluent infra.
+    pub fn fallback_fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.write_str(match self.without_payload() {
```

Comment:
There's no need to drop the payload here, is there?

Reply:
I suppose not! There was some vague thought in my head mumbling about "but multiple monomorphization..." but I don't think that matters actually.

Reply:
We're only going to call it with an empty payload, so it's probably fine. You could even implement it only for those, but 🤷.

Reply:
This method could be only on an `impl LayoutCalculatorError<()>` if you want to avoid multiple monomorphization.

Reply:
Doing that Feels Silly on some level, now that I've actually thought about it.